File: Training/OOPFundamentals/Java/Exercises/My Answers/06 Class Inheritance/src/Employee.java
public class Employee {
    private int idNumber;

    public int getIdNumber() {
        return this.idNumber;
    }

    public Employee(int id) {
        this.idNumber = id;
    }

    public boolean hasAdministratorRights() {
        return false;
    }
    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof Employee)) {return false;}


        return ((Integer) ((Employee) obj).getIdNumber()).equals(this.getIdNumber());
    }

}

File: Training/OOPFundamentals/C#/Exercises/after/ClassInheritance/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassInheritance
{
    public class Employee
    {
        private int _idNumber;

        public Employee(int id)
        {
            _idNumber = id;
        }

        public int IdNumber {
            get
            {
                return _idNumber;
            }
        }

       
        public virtual bool HasAdministratorRights()
        {
            return false;
        }
        public bool Equals(Employee a, Employee b)
        {
            return a.IdNumber == b.IdNumber ? true : false;
        }
    }
}
